src/services/gitlab/GitLabService.ts

```diff
@@ -0,0 +1,460 @@
+      user: { id: `${data.author.id}`, login: data.author.username, url: data.author.web_url },
+      id: data.id,
+      url: `${this.host}/projects/${owner}/${repo}/merge_requests/${pullRequestId}/notes`,
+      body: data.body,
+      createdAt: data.created_at.toString(),
+      updatedAt: data.updated_at.toString(),
+    };
+  }
+
+  async listContributors(owner: string, repo: string): Promise<Paginated<Contributor>> {
+    throw new Error('Method not implemented yet.');
+  }
+
+  async getContributorsStats(owner: string, repo: string): Promise<Paginated<ContributorStats>> {
+    throw new Error('Method not implemented yet.');
+  }
+
+  async getRepoContent(owner: string, repo: string, path: string): Promise<File | Symlink | Directory | null> {
+    throw new Error('Method not implemented yet.');
+  }
+
+  async getPullsDiffStat(owner: string, repo: string, prNumber: number): Promise<Lines> {
+    throw new Error('Method not implemented yet for GitLab.');
+  }
+
+  private async getUserInfo(owner: string) {
+    let userInfo;
+
+    try {
+      userInfo = await this.client.Users.getUser(owner);
+
+      return {
+        id: userInfo.data[0].id.toString(),
+        login: userInfo.data[0].username,
+        url: userInfo.data[0].web_url,
+      };
+    } catch (error) {
+      userInfo = await this.client.Users.getGroup(owner);
+
+      return {
+        id: userInfo.data.id.toString(),
+        login: userInfo.data.name,
+        url: userInfo.data.web_url,
+      };
+    }
+  }
+
+  private getPagination(pagination: PaginationGitLabCustomResponse): ServicePagination {
+    const hasNextPage = !!pagination.next;
+    const hasPreviousPage = !!pagination.previous;
+    const page = pagination.current;
+    const perPage = pagination.perPage;
+    let totalCount = pagination.total;
+
+    if (Number.isNaN(totalCount)) {
+      // If the number of resources is more than 10,000, the X-Total is not present in the response headers.
+      // https://docs.gitlab.com/ee/api/#other-pagination-headers
+      totalCount = 10000;
+    }
+
+    return { totalCount, hasNextPage, hasPreviousPage, page, perPage };
+  }
+
+  /**
+   * Debug GitLab request promise
+   */
+  private unwrap<T>(clientPromise: Promise<CustomAxiosResponse<T>>): Promise<CustomAxiosResponse<T>> {
+    return clientPromise
+      .then((response) => {
+        this.debugGitLabResponse(response);
+        return response;
+      })
+      .catch((error) => {
+        if (error.response) {
+          debug(`${error.response.status} => ${inspect(error.response.data)}`);
+        } else {
+          debug(inspect(error));
+        }
+        throw error;
+      });
+  }
+
+  /**
+   * Debug GitLab response
+   * - count API calls and inform about remaining rate limit
+   */
+  private debugGitLabResponse = <T>(response: CustomAxiosResponse<T>) => {
+    this.callCount++;
+    debug(`GitLab API Hit: ${this.callCount}. Remaining ${response.headers['RateLimit-Remaining']} hits.`);
```

Comment:
Has the Gitlab the header `RateLimit-Remaining` or any else?

Reply:
Yes, it has `RateLimit-Remaining` 
There are also values `RateLimit-Limit` and `RateLimit-Observed`